workspaces/controller/api/v1beta1/workspacekind_types.go

```diff
@@ -62,3 +74,288 @@ type WorkspaceKindList struct {
+	Remove []string          `json:"remove,omitempty"`
+}
+
+type ExtraEnv struct {
+	//+kubebuilder:example:="NB_PREFIX"
+	Name string `json:"name"`
+
+	//+kubebuilder:example:="{{ .PathPrefix }}"
+	Value string `json:"value"`
+}
+
+type KindOptions struct {
+	// imageConfig options determine the container image
+	ImageConfig ImageConfig `json:"ImageConfig"`
+
+	// podConfig options determine pod affinity, nodeSelector, tolerations, resources
+	PodConfig PodConfig `json:"podConfig"`
+}
+
+type ImageConfig struct {
+	// the id of the default image config for this WorkspaceKind
+	//+kubebuilder:example:="jupyter_scipy_171"
+	Default string `json:"default"`
+
+	// the list of image configs that are available to choose from
+	Values []ImageConfigValue `json:"values"`
+}
+
+type ImageConfigValue struct {
+	//+kubebuilder:example:="jupyter_scipy_171"
+	Id string `json:"id"`
+
+	Spawner  ImageConfigSpawner  `json:"spawner"`
+	Redirect ImageConfigRedirect `json:"redirect"`
+	Spec     ImageConfigSpec     `json:"spec"`
+}
+
+type ImageConfigSpawner struct {
+	//+kubebuilder:example:="jupyter-scipy:v1.7.0"
+	DisplayName string `json:"displayName"`
+
+	//+kubebuilder:example:="JupyterLab 1.7.0, with SciPy Packages"
+	Description string `json:"description"`
+
+	// if this ImageConfig should be hidden from the Workspace Spawner UI
+	//+kubebuilder:default:=false
+	Hidden bool `json:"hidden"`
+}
+
+type ImageConfigRedirect struct {
+	//+kubebuilder:example:="jupyter_scipy_171"
+	To string `json:"to"`
+
+	//+kubebuilder:example:=true
+	WaitForRestart bool `json:"waitForRestart"`
+}
+
+type ImageConfigSpec struct {
+	// the container image to use
+	//+kubeflow:example="docker.io/kubeflownotebookswg/jupyter-scipy:v1.7.0"
+	Image string `json:"image"`
+
+	// ports that the container listens on, currently, only HTTP is supported for `protocol`
+	//  if multiple ports are defined, the user will see multiple "Connect" buttons in a dropdown menu on the Workspace overview page
+	Ports []v1.EndpointPort `json:"ports"`
+}
+
+type PodConfig struct {
+	// the id of the default pod config
+	//+kubebuilder:example="small-cpu"
+	Default string `json:"default"`
+
+	// the list of pod configs that are available
+	Values []PodConfigValue `json:"values"`
+}
+
+type PodConfigValue struct {
+	//+kubebuilder:example="small-cpu"
+	Id string `json:"id"`
+
+	Spawner  PodConfigSpawner  `json:"spawner"`
+	Redirect PodConfigRedirect `json:"redirect"`
+	Spec     PodConfigSpec     `json:"spec"`
+}
+
+type PodConfigSpawner struct {
+	//+kubebuilder:example:="Small CPU"
+	DisplayName string `json:"display_name,omitempty"`
+
+	//+kubebuilder:example:="Pod with 1 CPU, 2 GB RAM, and 1 GPU"
+	Description string `json:"description,omitempty"`
+
+	//+kubebuilder:example:=false
+	Hidden bool `json:"hidden,omitempty"`
+}
+
+type PodConfigRedirect struct {
+	//+kubebuilder:example:="large-cpu"
+	To string `json:"to"`
+
+	//+kubebuilder:example:=true
+	WaitForRestart bool `json:"waitForRestart"`
```

Comment:
This should default to true and be optional:

```suggestion
	//+kubebuilder:default:=true
	WaitForRestart bool `json:"waitForRestart,omitempty"`
```